packages/roosterjs-content-model/lib/modelApi/selection/iterateSelections.ts

```diff
@@ -103,9 +105,63 @@ export function iterateSelections(
                 });
 
                 if (segments.length > 0) {
+                    if (segments.length == 1 && segments[0].segmentType == 'SelectionMarker') {
```

Comment:
iterateSelections() is a common function, it should not do any feature-specified logic, instead it will just invoke the callback function with raw selection info.

You can move this code out to be a util function.

I expect you finally export a function like:

funciton adjustSelectionForWord(model: ContentModelDocument) {
 // Here you can first get selections using iterateSelections() or some other util function, check if selection is collapsed, then adjust it by splitting existing text segment if any, like you have already done.
// Then after running this function, the selections are changed to the splitted segments. Then we just need to run regular format API to do formatting as if those words was selected by user.
}